neuralop/layers/spectral_convolution.py

```diff
@@ -335,15 +335,23 @@ def __init__(
 
         tensor_kwargs = decomposition_kwargs if decomposition_kwargs is not None else {}
 
-        # Create/init spectral weight tensor
-
+        # Determine weight dtype based on fno_block_precision
+        self.weight_dtype = {
+            "full": torch.cfloat,
+            "half": torch.chalf,
+            "mixed": torch.chalf
+        }.get(fno_block_precision, torch.cfloat)
```

Comment:
Looks good though, wouldn't we want to fail loudly rather than silently use cfloat in case of wrong type? 
I'd first define the dict then catch any KeyError and write an explicit message for users to help them fix the issue.